src/Common/src/System/Net/IPv6AddressHelper.Common.cs

```diff
@@ -411,6 +411,13 @@ internal static void Parse(ReadOnlySpan<char> address, Span<ushort> numbers, int
                 int toIndex = NumberOfLabels - 1;
                 int fromIndex = index - 1;
 
+                // if fromIndex and toIndex are the same mean that "zero bits" are already in the correct place
+                // it happens for leading and trailing compression
+                if (fromIndex == toIndex)
+                {
+                    return;
+                }
```

Comment:
Nit: could you instead make this:
```C#
if (fromIndex != toIndex)
{
    for (int i = ...)
    {
        ...
    }
}
```
The `return` here is a bit hidden.

Reply:
I did it in first version 😄 ok.

Reply:
done